autofit/mapper/operator.py

```diff
@@ -609,7 +619,19 @@ def __init__(self, scale, inv_scale=None):
         if inv_scale is not None:
             self._finv_scale = np.ravel(inv_scale)
 
-        self._ldim = len(self.scale.shape)
+        self._ldim = self.scale.ndim
+
+    def __getitem__(self, index):
+        if index[:self.ldim] == index[self.ldim:]:
+            return DiagonalMatrix(
+                self.scale[index[:self.ldim]]
+            )
+        else:
+            raise NotImplementedError()
```

Comment:
It's work adding a message explaining this.

Also this should probably be a ValueError